internal/internal_test.go

```diff
@@ -1114,6 +1115,23 @@ func TestGetResponseHeaders(t *testing.T) {
 	if err == nil {
 		t.Fatal("Expected error but got nil")
 	}
+
+	testHeaders = make(map[string]interface{})
+	testHeaders["foo"] = "bar"
+	testHeaders2 := make(map[string]interface{})
+	testHeaders2["foo"] = "baz"
+	testHeadersSlices := make([]interface{}, 0)
+	testHeadersSlices = append(testHeadersSlices, testHeaders)
+	testHeadersSlices = append(testHeadersSlices, testHeaders2)
+	input["headers"] = testHeadersSlices
```

Comment:
It's probably just a matter of preference -- what do you think about this? 

```suggestion
	input["headers"] = []interface{}{
		map[string]interface{}{
			"foo": "bar",
		},
		map[string]interface{}{
			"foo": "baz",
		},
	}
```

I find it slightly easier to read.